### ISSUE TITLE
Make GRPCAsyncRequestStreamWriter's finish() sync

### ISSUE THREAD

**Issue by schwmi** [grpc/grpc-swift#1503]
Currently the finish() method on `GRPCAsyncRequestStreamWriter` is marked as `async` + `throws` but there is no internal need for that. As API consumer a `sync` finish() method would make it slightly easier to cleanup after usage. If you are ok with this change I can open a PR.

Current state:
```
  /// Finish the request stream for the RPC. This must be called when there are no more requests to be sent.
  public func finish() async throws {
    self.asyncWriter.finish()
  }
```

Suggestion:
```
  /// Finish the request stream for the RPC. This must be called when there are no more requests to be sent.
  public func finish() {
    self.asyncWriter.finish()
  }
```

**Comment by FranzBusch** [comment]
Thanks for opening this issue @schwmi. Sadly, we cannot change the method signature of `finish()` since this would be an API breaking change. However, we ought to be able to introduce a new method next to the async `finish()` method that is sync.

**Comment by schwmi** [comment]
Thanks for the response - makes sense. 
NIO has on it's EventLoopGroup a `shutdownGracefully` and `syncShutdownGracefully`. Could also add a `syncFinish() beside the `finish()` method. 
Or as this is not the most pressing change you could also keep the signature change for `finish()` in the backlog for the next major version bump (if any is planned).

**Comment by FranzBusch** [comment]
I think we are fine having two finish methods that are named identical. We might even go as far and mark the async variant as `deprecated`. Feel free to open a PR and we can further discuss it there and @glbrntt can also way in.

**Comment by glbrntt** [comment]
Resolved by #1504
